catalog/imaging/imaging-postprocess-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,556 @@
+
+        new NitfParserInputFlowImpl()
+            .inputStream(inputStream)
+            .allData()
+            .forEachImageSegment(
+                segment -> {
+                  if (bufferedImage.get() == null) {
+                    BufferedImage bi =
+                        imageSegmentFunction.apply(new ImmutablePair<>(segment, renderer));
+                    if (bi != null) {
+                      bufferedImage.set(bi);
+                    }
+                  }
+                })
+            .end();
+      } finally {
+        IOUtils.closeQuietly(inputStream);
+        available.release();
+      }
+    }
+
+    return bufferedImage.get();
+  }
+
+  private void addThumbnailToMetacard(Metacard metacard, BufferedImage bufferedImage) {
+    try {
+      byte[] thumbnailImage = scaleImage(bufferedImage, THUMBNAIL_WIDTH, THUMBNAIL_HEIGHT);
+
+      if (thumbnailImage.length > 0) {
+        metacard.setAttribute(new AttributeImpl(Core.THUMBNAIL, thumbnailImage));
+      }
+    } catch (IOException e) {
+      LOGGER.debug(e.getMessage(), e);
+    }
+  }
+
+  private ProcessResource createDerivedImage(
+      String qualifier, BufferedImage image, Metacard metacard, int maxWidth, int maxHeight) {
+    try {
+      byte[] overviewBytes = scaleImage(image, maxWidth, maxHeight);
+      InputStream overviewBytesInputStream = new ByteArrayInputStream(overviewBytes);
+
+      ProcessResource processResource =
+          new ProcessResourceImpl(
+              metacard.getId(),
+              overviewBytesInputStream,
+              IMAGE_JPEG,
+              buildDerivedImageTitle(metacard.getTitle(), qualifier, JPG),
+              overviewBytes.length,
+              qualifier);
+
+      ((ProcessResourceImpl) processResource).markAsModified();
+      addDerivedResourceAttribute(metacard, processResource);
+
+      return processResource;
+    } catch (IOException e) {
+      LOGGER.debug(e.getMessage(), e);
+    }
+
+    return null;
+  }
+
+  private ProcessResource createOriginalImage(BufferedImage image, Metacard metacard) {
+
+    try {
+      byte[] originalBytes = renderToJpeg2k(image);
+
+      InputStream originalBytesInputStream = new ByteArrayInputStream(originalBytes);
+
+      ProcessResource processResource =
+          new ProcessResourceImpl(
+              metacard.getId(),
+              originalBytesInputStream,
+              IMAGE_JPEG2K,
+              buildDerivedImageTitle(metacard.getTitle(), ORIGINAL, JP2),
+              originalBytes.length,
+              ORIGINAL);
+
+      ((ProcessResourceImpl) processResource).markAsModified();
+
+      addDerivedResourceAttribute(metacard, processResource);
+
+      return processResource;
+
+    } catch (IOException e) {
+      LOGGER.debug(e.getMessage(), e);
+    }
+
+    return null;
+  }
+
+  @VisibleForTesting
+  static String buildDerivedImageTitle(String title, String qualifier, String extension) {
+    String rootFileName = FilenameUtils.getBaseName(title);
```

Comment:
❓Since the title could be something else than a file name, is there a reason why we're calling `getBaseName(title)` here? Doesn't L444 already take care of what we need?